### PR TITLE
Implement DataAuditResult type

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: v2/refs/marshal.go [hunk 1]
```diff
@@ -123,6 +133,23 @@ func (o *ObjectID) StableSize() int {
 	return proto.BytesSize(objectIDValField, o.val)
 }
 
+// ObjectIDNestedListSize returns byte length of nested
+// repeated ObjectID field with fNum number.
+func ObjectIDNestedListMarshal(fNum int64, buf []byte, ids []*ObjectID) (off int, err error) {
```
Comment (alexvanin) [inline v2/refs/marshal.go:138]: Can we reuse this function in `searchResponseBody.StableMarshal()`? It also marshals list of object IDs.

File: v2/refs/marshal.go [hunk 2]
```diff
@@ -123,6 +133,23 @@ func (o *ObjectID) StableSize() int {
 	return proto.BytesSize(objectIDValField, o.val)
 }
 
+// ObjectIDNestedListSize returns byte length of nested
```
Comment (alexvanin) [inline v2/refs/marshal.go:136]: Fix comment, it is from `ObjectIDNestedListSize()`.

File: v2/refs/convert.go [hunk 3]
```diff
@@ -76,6 +76,34 @@ func ObjectIDFromGRPCMessage(m *refs.ObjectID) *ObjectID {
...
+	idsV2 := make([]*refs.ObjectID, 0, len(ids))
+
+	for i := range ids {
+		idsV2 = append(idsV2, ObjectIDToGRPCMessage(ids[i]))
+	}
+
+	return idsV2
+}
+
+func ObjectIDListFromGRPCMessage(idsV2 []*refs.ObjectID) []*ObjectID {
```
Comment (alexvanin) [inline v2/refs/convert.go:93]: Can we reuse it in `SearchResponseBodyToGRPCMessage` and `SearchResponseBodyFromGRPCMessage`?

File: pkg/audit/result.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+		MarshalJSON()
+}
+
+// UnmarshalJSON decodes Result from protobuf JSON format.
+func (r *Result) UnmarshalJSON(data []byte) error {
+	return (*audit.DataAuditResult)(r).
+		UnmarshalJSON(data)
+}
+
+// AuditEpoch returns returns epoch number when the Data Audit was conducted.
```
Comment (alexvanin) [inline pkg/audit/result.go:60]: returns returns

File: pkg/audit/result.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+// PassNodes returns list of storage node public keys that
+// passed at least one PDP.
+func (r *Result) PassNodes() [][]byte {
+	return (*audit.DataAuditResult)(r).
+		GetPassNodes()
+}
+
+// SetPassNodes returns list of storage node public keys that
```
Comment (alexvanin) [inline pkg/audit/result.go:239]: s/returns/sets

File: pkg/audit/result.go [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+// FailNodes returns list of storage node public keys that
+// failed at least one PDP.
+func (r *Result) FailNodes() [][]byte {
+	return (*audit.DataAuditResult)(r).
+		GetFailNodes()
+}
+
+// SetFailNodes returns list of storage node public keys that
```
Comment (alexvanin) [inline pkg/audit/result.go:253]: s/returns/sets

File: pkg/audit/result.go [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+
+// Fail returns number of sampled objects under audit stored
+// in a way not confirming placement policy or not found at all
+// when checking PoP.
+func (r *Result) Fail() uint32 {
+	return (*audit.DataAuditResult)(r).
+		GetFail()
+}
+
+// SetFail returns number of sampled objects under audit stored
```
Comment (alexvanin) [inline pkg/audit/result.go:224]: s/returns/sets

**Review by alexvanin — APPROVED** [review]

File: v2/refs/marshal.go (hunk 1 above)
Comment (cthulhu-rider) [reply v2/refs/marshal.go:138]: Yes we can

File: v2/refs/convert.go (hunk 3 above)
Comment (cthulhu-rider) [reply v2/refs/convert.go:93]: Yes we can

File: v2/refs/marshal.go (hunk 2 above)
Comment (cthulhu-rider) [reply v2/refs/marshal.go:136]: Fixed

File: pkg/audit/result.go (hunk 7 above)
Comment (cthulhu-rider) [reply pkg/audit/result.go]: Fixed

File: pkg/audit/result.go (hunk 6 above)
Comment (cthulhu-rider) [reply pkg/audit/result.go]: Fixed

File: pkg/audit/result.go (hunk 5 above)
Comment (cthulhu-rider) [reply pkg/audit/result.go]: Fixed

File: pkg/audit/result.go (hunk 4 above)
Comment (cthulhu-rider) [reply pkg/audit/result.go]: Fixed

**Pull request merged**